File: src/removeDuplicates/removeDuplicates.go
package removeDuplicates

//https://leetcode-cn.com/problems/remove-duplicates-from-sorted-array/solution/
func removeDuplicates(nums []int) int {
	var res int

	for i, _ := range nums {
		if i > 0 && nums[i] == nums[i - 1] {
			continue
		}

		if nums[res] != nums[i] {
			res++
			nums[res] = nums[i]
		}
	}

	return res + 1
}

File: c/src/removeDuplicates/removeDuplicates.c
#include "removeDuplicates.h"
#include "stdio.h"

int removeDuplicates(int* nums, int numsSize) {
    int res = 1, i = 1;

    if (numsSize == 1) {
        return res;
    }

    for (;i < numsSize; i++) {
        if (nums[i] == nums[i - 1]) {
            continue;
        }

        nums[res] = nums[i];
        res++;
    }

    return res;
}